packages/core/src/render3/assert.ts

```diff
@@ -27,10 +27,9 @@ export function assertTNodeForLView(tNode: TNode, lView: LView) {
 
 export function assertTNodeForTView(tNode: TNode, tView: TView) {
   assertTNode(tNode);
-  tNode.hasOwnProperty('tView_') &&
-      assertEqual(
-          (tNode as any as {tView_: TView}).tView_, tView,
-          'This TNode does not belong to this TView.');
+  if (!tView.data.includes(tNode)) {
+    throwError('This TNode does not belong to this TView.');
+  }
```

Comment:
I think we can limit the area where we search for TNodes to something like this:
https://github.com/angular/angular/blob/125fe19086e3c845816d46c990899fd3edc1ebbe/packages/core/src/render3/context_discovery.ts#L220

(i.e. a TNode can be located after HEADER_OFFSET and beforefirst binding)

```suggestion
  let hasTNode = false;
  for (let i = HEADER_OFFSET; i < tView.bindingStartIndex; i++) {
    if (tView.data[i] === tNode) {
      hasTNode = true;
      break;
    }
  }
  if (!hasTNode) {
    throwError('This TNode does not belong to this TView.');
  }
```

Reply:
Your suggestion breaks 18 tests. Are you sure about this ?

Reply:
Yes, we should skip at least the header part:
```suggestion
  let hasTNode = false;
  const tData = tView.data;
  for (let i = HEADER_OFFSET; i < tData.length; i++) {
    if (tData[i] === tNode) {
      hasTNode = true;
      break;
    }
  }
  if (!hasTNode) {
    throwError('This TNode does not belong to this TView.');
  }
```
I've checked the code and in case of i18n (I guess this is what was broken?), we also create TNodes and store them in a section called "expando", which doesn't have a constrained length. The "expando" section can also contain other info (such as matched directive defs, etc). In this case, we can just iterate over the entire array, starting from the `HEADER_OFFSET` position.

Reply:
Thanks for the insight  !